Flaky test: sshreceiver/TestScraper

### Component(s)

receiver/sshcheck

### Describe the issue you're reporting

See https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4164947978/jobs/7207502717#step:9:4149

```
=== RUN   TestScraper/metrics_golden
    scraper_test.go:134: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver/scraper_test.go:134
        	Error:      	Received unexpected error:
        	            	resource "map[]": scope "otelcol/sshcheckreceiver": number of metrics doesn't match expected: 2, actual: 3
        	Test:       	TestScraper/metrics_golden
=== RUN   TestScraper/metrics_golden_sftp
    scraper_test.go:134: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver/scraper_test.go:134
        	Error:      	Received unexpected error:
        	            	resource "map[]": scope "otelcol/sshcheckreceiver": number of metrics doesn't match expected: 4, actual: 5
        	Test:       	TestScraper/metrics_golden_sftp
=== RUN   TestScraper/cannot_authenticate
    scraper_test.go:134: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver/scraper_test.go:134
        	Error:      	Received unexpected error:
        	            	resource "map[]": scope "otelcol/sshcheckreceiver": number of metrics doesn't match expected: 2, actual: 3
        	Test:       	TestScraper/cannot_authenticate
=== RUN   TestScraper/invalid_endpoint
    scraper_test.go:134: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sshcheckreceiver/scraper_test.go:134
        	Error:      	Received unexpected error:
        	            	resource "map[]": scope "otelcol/sshcheckreceiver": number of metrics doesn't match expected: 2, actual: 3
        	Test:       	TestScraper/invalid_endpoint
--- FAIL: TestScraper (4.89s)
```

--- a/receiver/sshcheckreceiver/scraper_test.go
+++ b/receiver/sshcheckreceiver/scraper_test.go
@@ -16,6 +16,7 @@ package sshcheckreceiver // import "github.com/open-telemetry/opentelemetry-coll
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"path/filepath"
 	"testing"
@@ -53,7 +54,7 @@ func setupSSHServer(t *testing.T) *opensshContainer {
 			"PASSWORD_ACCESS": "true",
 		},
 		ExposedPorts: []string{"2222/tcp"},
-		WaitingFor:   wait.ForExposedPort(),
+		WaitingFor:   wait.ForListeningPort("2222/tcp"),
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
@@ -64,9 +65,17 @@ func setupSSHServer(t *testing.T) *opensshContainer {
 		Started:          true,
 	})
 	require.NoError(t, err)
+	require.NotNil(t, container)
 
-	endpoint, err := container.PortEndpoint(ctx, "2222", "")
+	mappedPort, err := container.MappedPort(ctx, "2222")
+	print(mappedPort)
 	require.NoError(t, err)
+
+	hostIP, err := container.Host(ctx)
+	require.Nil(t, err)
+
+	endpoint := fmt.Sprintf("%s:%s", hostIP, mappedPort.Port())
+
 	t.Log("endpoint: ", endpoint)
 	return &opensshContainer{Container: container, Endpoint: endpoint}
 }

--- a/receiver/sshcheckreceiver/scraper_test.go
+++ b/receiver/sshcheckreceiver/scraper_test.go
@@ -68,7 +68,6 @@ func setupSSHServer(t *testing.T) *opensshContainer {
 	require.NotNil(t, container)
 
 	mappedPort, err := container.MappedPort(ctx, "2222")
-	print(mappedPort)
 	require.NoError(t, err)
 
 	hostIP, err := container.Host(ctx)

